client.go

```diff
@@ -162,12 +172,32 @@ var h2Transport = http2.Transport{
 // NewClient creates a RESTful client instance.
 // The instance has a semi-permanent transport TCP connection.
 func NewClient() *Client {
+	return NewClientWInterface("")
+}
+
+// NewClientWInterface creates a RESTful client instance with binded to that interface.
+// The instance has a semi-permanent transport TCP connection.
+func NewClientWInterface(theInterface string) *Client {
 	t := http.DefaultTransport.(*http.Transport).Clone()
 	t.MaxIdleConns = 100
 	t.MaxConnsPerHost = 100
 	t.MaxIdleConnsPerHost = 100
 	dialer := &net.Dialer{Timeout: 2 * time.Second, KeepAlive: 30 * time.Second}
-	t.DialContext = dialer.DialContext
+	if theInterface != "" {
+		IPs := getIpFromInterface(theInterface)
+		t.DialContext = func(ctx context.Context, network, addr string) (net.Conn, error) {
+			dialer.LocalAddr = IPs.IPv4
```

Comment:
I'd check if the pointer was nil.

Reply:
If localAddr is nil, a local address is automatically chosen.